rule.proto

```diff
@@ -0,0 +1,81 @@
+syntax = "proto2";
+package dv360_excel_plugin;
+
+// A message that represents a scheduled rule for manipulating DV360 entities.
+message Rule {
+  // required
+  // The name of the rule, set by the user.
+  optional string name = 1;
+
+  // required
+  // The id that uniquely identified each rule.
+  optional int32 id = 2;
+
+  // required
+  optional Action action = 3;
+
+  // required
+  optional Schedule schedule = 4;
+}
+
+// A message representing the action applied to a DV360 entity.
+message Action {
+  enum Type {
+    UNSPECIFIED_TYPE = 0;
+    CHANGE_LINE_ITEM_STATUS = 1;
+  }
+
+  // required
+  // The type of action.
+  optional Type type = 1;
+
+  // required
+  // The parameters used for each type of action.
+  oneof params {
+    ChangeLineItemStatusParams change_line_item_status_params = 2;
+  }
+}
+
+// The action parameters to change line items' status.
+message ChangeLineItemStatusParams {
+  // required
+  // The ID of line items to which the action applies.
+  repeated int64 line_item_ids = 1;
+
+  enum Status {
+    UNSPECIFIED_STATUS = 0;
+    ACTIVE = 1;
+    PAUSED = 2;
+  }
+
+  // required
+  // The status of which the line item will be changed to.
+  // Can be set to either active or paused.
+  optional Status status = 2;
+}
+
+// required
+// A message representing the schedule the rule will be run.
+message Schedule {
+  enum Type {
+    UNSPECIFIED_SCHEDULE = 0;
+    REPEATING = 1;
+  }
+
+  // required
+  // The rule can be scheduled for either repeating or one-time.
+  optional Type type = 1;
+
+  // The parameters for a repeating rule.
+  message RepeatingParams {
+    // required
+    // The cron expression representing the repeating schedule.
+    optional string cron_expression = 1;
```

Comment:
is this directly consumed by Scheduler? If the format is like the cron job definition in Linux '* * * * * *` and input by user, seems it is not user friendly.

Reply:
Yes, Scheduler takes cron expressions. I was thinking the UI could look something like this and just transform it into a cron expression on the client side:

<img width="1155" alt="Screen Shot 2020-07-29 at 3 18 23 PM" src="https://user-images.githubusercontent.com/67433172/88859659-031dfb80-d1af-11ea-8dc9-658cdc7b11a3.png">

Should I separate it out to individual fields?

Reply:
I think doing conversion in client side works.